### PR TITLE
test(python): Add parameterized-scan-tests

### PR DESCRIPTION
**Pull request by mickvangelderen**

### CODE REVIEW

File: py-polars/tests/unit/io/test_scan.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+def _scan(file_path: Path, schema: SchemaDict | None = None) -> pl.LazyFrame:
+    suffix = file_path.suffix
+    if suffix == ".ipc":
+        return pl.scan_ipc(file_path)
+    if suffix == ".parquet":
+        return pl.scan_parquet(file_path)
+    if suffix == ".csv":
+        return pl.scan_csv(file_path, schema=schema)
+    msg = f"Unknown suffix {suffix}"
+    raise NotImplementedError(msg)
```
Comment (mickvangelderen) [inline py-polars/tests/unit/io/test_scan.py:26]: Does polars already provide a uniform `scan` interface?

File: py-polars/tests/unit/io/test_scan.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+def _write(df: pl.DataFrame, file_path: Path) -> None:
+    suffix = file_path.suffix
+    if suffix == ".ipc":
+        return df.write_ipc(file_path)
+    if suffix == ".parquet":
+        return df.write_parquet(file_path)
+    if suffix == ".csv":
+        return df.write_csv(file_path)
+    msg = f"Unknown suffix {suffix}"
+    raise NotImplementedError(msg)
```
Comment (mickvangelderen) [inline py-polars/tests/unit/io/test_scan.py:38]: Same for `write`

**Review by ritchie46 — APPROVED** [review]

File: py-polars/tests/unit/io/test_scan.py (hunk 1 above)
Comment (stinodego) [reply py-polars/tests/unit/io/test_scan.py:26]: This doesn't exist yet, at least not on the Python side.

File: py-polars/tests/unit/io/test_scan.py (hunk 2 above)
Comment (stinodego) [reply py-polars/tests/unit/io/test_scan.py:38]: Also doesn't exist.

**Pull request merged**